src/facade.ts

```diff
@@ -0,0 +1,39 @@
+import type * as AppMap from "./AppMap";
+
+import { recording, start } from "./recorder";
+import {
+  disableGlobalRecording,
+  startCodeBlockRecording,
+  stopCodeBlockRecording,
+} from "./recorderControl";
+import Recording from "./Recording";
+
+// Since _this_ module is loaded, we'll do code block recording only.
+recording.abandon();
+disableGlobalRecording();
+
+export function record(block: () => unknown): AppMap.AppMap | undefined {
+  start(new Recording("block", "block", new Date().toISOString()));
```

Comment:
The record function should probably check if the code is instrumented and throw otherwise, telling the user to run with `appmap-node`. Otherwise it will just mysteriously fail to produce anything. (Add a test case for that!)